### DISCUSSION TITLE
Migrations contains extra statements when cascade all is used

### DISCUSSION THREAD

**Discussion by PrabuN19** [mikro-orm/mikro-orm#3204] [Q&A]
**Describe the bug**
When cascade all option is used in the entity class, and migrations are generated.
Every time, the generated migrations contain extra sql statements which are already applied.

**Example**

**Entity class:**

```typescript
@Entity()
export class AppliedJobs {
  [EntityRepositoryType]?: AppliedJobsRepository;

  @ManyToOne({ entity: () => User, primary: true })
  user: User;

  @ManyToOne({ entity: () => Job, primary: true, cascade: [Cascade.ALL] })
  job: Job;

  @Property()
  createdAt = new Date();

  @Property({ onUpdate: () => new Date() })
  updatedAt = new Date();

  [PrimaryKeyType]: [number, number];

  constructor(user: User, job: Job) {
    this.user = user;
    this.job = job;
  }
}
 ```

**Existing migration**

```typescript
 this.addSql('alter table "applied_jobs" drop constraint if exists "applied_jobs_job_id_foreign";');
 this.addSql('alter table "applied_jobs" add constraint "applied_jobs_job_id_foreign" foreign key ("job_id") references "job" ("id") on update cascade on delete cascade;');
```

**New migration (contains extra statements from previous migration)**

```typescript
 this.addSql('alter table "user" add constraint "user_role_check" check ("role" in (\'student\', \'admin\'));');

 this.addSql('alter table "applied_jobs" drop constraint if exists "applied_jobs_job_id_check";');
 this.addSql('alter table "applied_jobs" alter column "job_id" type int4 using ("job_id"::int4);');
 this.addSql('alter table "applied_jobs" alter column "job_id" drop not null;'); 
```

**Comment by B4nan** [comment]
What version are you on? Please dont ignore issue template. 

Try using latest dev version, it adds extensive logging to the schema comparator so you can see why it generates new diff.

edit: actually i am just releasing things to stable, so you can also use v5.2.0 for that, will be available in few minuts

Note that your date columns are wrongly defined unless you use ts-morph. But hard to guess, you havent provided any details about your setup, not about versions 🤷 I need to see all the entities connected to your problem, notably the diff you see is probably caused by the `Job` entity and `User` entity.
